src/languages/stan.js

```diff
@@ -504,16 +345,38 @@ export default function(hljs) {
       $pattern: hljs.IDENT_RE,
       title: BLOCKS,
       keyword: STATEMENTS.concat(VAR_TYPES).concat(SPECIAL_FUNCTIONS),
-      built_in: FUNCTIONS
+      built_in: FUNCTIONS.concat(DISTRIBUTIONS).concat(expanded),
+      punctuation: DELIMIT
     },
     contains: [
       hljs.C_LINE_COMMENT_MODE,
       INCLUDE,
       hljs.HASH_COMMENT_MODE,
       BLOCK_COMMENT,
+     {
+       className: 'built_in',
+       variants: [
+         {
+          begin: /(?<=~)\s+(\w+)/
+         },
+         {
+          begin: /(?<=\+=)\s+(\w+(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf))/
+         }
+        ] 
+     },
+     {
+      className: 'symbol',
+      begin: /\s+(pi|e|sqrt2|log2|log10)(?=\()/,
+      relevance: 0
+     },
+     {
+       begin: /(?<=\))\s*T(?=\s*\[)/,
```

Comment:
`(?<`

We cannot use regex look-behind anywhere because Safari does not yet support look-behind, sorry.

Reply:
Ah, I'll comment this out and can revisit once Safari gets look-behind. Not sure how to get this to work otherwise
```
vector[N] T[N); // do not highlight T
X ~ normal(mu, sigma) T[1, ]; // highlight T 
```

Reply:
You can use multi-match sometimes, but it will **consume** the prior content (so you can't highlight it, or it gets very complicated)... ie there is never a way to look-behind, you're always looking forward... not sure that would help us here though so perhaps better to just wait as you suggest.

Reply:
You can't do this, you can't run raw JS code on the input.

As I mentioned before you can use multi-matchers to achieve things *similar* to this (match groups)... see the docs for `beginScope` and `endScope` which can be keyed just like TM grammars, etc... as to whether it'd be allowed it would depend on how complex the actual implementation ended up being.